net-utils/src/lib.rs

```diff
@@ -391,22 +386,52 @@ pub fn is_host_port(string: String) -> Result<(), String> {
     parse_host_port(&string).map(|_| ())
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct SocketConfig {
-    pub reuseport: bool,
+    reuseport: bool,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    recv_buffer_size: Option<usize>,
+    #[cfg(not(any(windows, target_os = "ios")))]
+    send_buffer_size: Option<usize>,
 }
 
-impl Default for SocketConfig {
-    #[allow(clippy::derivable_impls)]
-    fn default() -> Self {
-        Self { reuseport: false }
+impl SocketConfig {
+    pub fn reuseport(mut self, reuseport: bool) -> Self {
+        self.reuseport = reuseport;
+        self
     }
-}
 
-#[cfg(any(windows, target_os = "ios"))]
-fn udp_socket(_reuseaddr: bool) -> io::Result<Socket> {
-    let sock = Socket::new(Domain::IPV4, Type::DGRAM, None)?;
-    Ok(sock)
+    /// Sets the receive buffer size for the socket.
+    ///
+    /// **Note:** On Linux the kernel will double the value you specify.
+    /// For example, if you specify `16MB`, the kernel will configure the
+    /// socket to use `32MB`.
+    /// See: https://man7.org/linux/man-pages/man7/socket.7.html: SO_RCVBUF
+    // allow here to supress unused warnings from windows/ios builds
+    #[allow(unused_mut, unused_variables)]
+    pub fn recv_buffer_size(mut self, size: usize) -> Self {
+        #[cfg(not(any(windows, target_os = "ios")))]
```

Comment:
similarly here (and perhaps other places in the code) there is no harm keeping the code the same across platforms.
so `#[cfg(...)]` here is probably unnecessary.